### PR TITLE
Add new user_admin role

### LINKED ISSUE #2574: Grid admin should be able to invite users to a grid
Currently it requires that master_admin first invites user to master.

### PR DESCRIPTION
**Pull request by SpComb**
Fixes #2574

* New `user_admin` role can use the `kontena master user` commands that were previously limited to `master_admin`:
  * `kontena master user invite` -> `POST /oauth2/authorize` -> `Users::Invite` -> `user.can_create?(User)`
  * `kontena master user list` -> `GET /v1/users` -> `user.can_read?(User)`
  * `kontena master user remove` -> `DELETE /v1/users/:email` -> `Users::Remove` -> `user.can_delete?(User)`
* The `user_admin` role can also use the `kontena grid user` commands if they have been added to the grid, and have access to the `/v1/grids/:name/...` APIs:
  * `kontena grid user add` -> `POST /v1/grids/:name/users` -> `Grids::AssignUser` -> `user.can_assign?(self.user, {to: self.grid})`
  * `kontena grid user remove` -> `DELETE /v1/grids/:name/users/:email` -> `Grids::UnassignUser` -> `???` (XXX: #2578)

## TODO

* Should the `user_admin` also be able to use the `kontena master user role add/remove` commands to grant the `grid_admin` and `user_admin` roles (but not `master_admin`)? Should there also be a `role_admin` role?

### CODE REVIEW

**Comment by SpComb** [comment]
Tested this to work, with a `user_admin` user able to invite a different user, and add to grid.

File: server/db/migrations/26_create_user_admin_role.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class CreateUserAdminRole < Mongodb::Migration
```
Comment (SpComb) [inline server/db/migrations/26_create_user_admin_role.rb]: The migration version here collides with #2504

File: docs/using-kontena/users.md [hunk 2]
```diff
@@ -4,23 +4,41 @@ title: Users
...
+
+### `users_admin`
+  * Invite and remove users
+  * Manage grid users (only user grids)
+
+### `grid_admin`
+
+  * Manage grid (update)
+  * Manage grid users
+  * Manage grid host nodes (create, reset tokens)
```
Comment (SpComb) [inline docs/using-kontena/users.md]: BTW: `kontena node rm` should probably also require `grid admin`? What about the `kontena node update` labels, and  #2306 availability?

Comment (jakolehm) [inline docs/using-kontena/users.md]: yes

Comment (SpComb) [inline docs/using-kontena/users.md]: Issue #2612

**Pull request merged**